components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSUtils.java

```diff
@@ -124,16 +124,46 @@ public static void setTransportHeaders(MessageContext msgContext, Message messag
                 }
 
             } else if (JMSConstants.JMS_EXPIRATION.equals(name)) {
-                message.setJMSExpiration(
-                    Long.parseLong((String) headerMap.get(JMSConstants.JMS_EXPIRATION)));
+                Object o = headerMap.get(JMSConstants.JMS_EXPIRATION);
```

Comment:
Shall we use meaningful name for the variable

Reply:
Yes. Better to avoid single letter variables